controller.py

```diff
@@ -78,6 +81,24 @@ def authenticated_client_from_request(_db, required=True):
     return INVALID_CREDENTIALS
 
 
+def collection_from_details(_db, client, collection_details):
+    if not (client and isinstance(client, IntegrationClient)):
+        return None
+
+    if collection_details:
+        # A DataSource may be sent for collections with the
+        # ExternalIntegration.OPDS_IMPORT protocol.
+        data_source_name = request.args.get('data_source')
```

Comment:
To reflect this change I think you need to add `{?data_source}` to the appropriate links in the OPDS 2 document generated by `opds_catalog`.